model_analyzer/record/metrics_mapper.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .gpu_free_memory import GPUFreeMemory
+from .gpu_used_memory import GPUUsedMemory
+from .gpu_utilization import GPUUtilization
+from .cpu_used_memory import CPUUsedMemory
+from .cpu_free_memory import CPUFreeMemory
+from .perf_throughput import PerfThroughput
+from .perf_latency import PerfLatency
+
+
+class MetricsMapper:
+    """
+    Class maps metrics to record types
+    """
+
+    tag_to_metric = {
+        "Throughput": PerfThroughput,
+        "Latency": PerfLatency,
+        "GPU Memory Usage": GPUUsedMemory,
+        "GPU Memory Available": GPUFreeMemory,
+        "GPU Utilization": GPUUtilization,
+        "RAM Usage": CPUUsedMemory,
+        "RAM Available": CPUFreeMemory
+    }
+
+    def __init__(self, tags):
+        """
+        Parameters
+        ----------
+        tags : list of str
+            Human readable names for the 
+            metrics to monitor. They correspond
+            to actual record types.
+        """
+
+        self._tags = tags
+
+    def get_monitoring_metrics(self):
```

Comment:
Also, I think it would be great to make this class a semi-factor class that doesn't need to be instantiated. How about making this method static and getting an argument for the list of metrics?

Reply:
Agreed.